src/armnn.js

```diff
@@ -0,0 +1,625 @@
+
+    getAttr(descriptor, key) {
+        if (typeof descriptor[key] == "undefined")
+            return "undefined";
+
+        let values = descriptor[key]();
+        if (Array.isArray(values)) {
+            return values.join(", ");
+        }
+        else {
+            return values;
+        }
+    }
+
+    packAttr(layer, attr) {
+        let descriptor = this.getDescriptor(layer);
+
+        let key  = attr.src;
+        let type = attr.src_type;
+
+        if (typeof type != "undefined") {
+            let value = this.getAttr(descriptor, key);
+            if (typeof armnn.schema[type + "Name"] != "undefined")
+                return armnn.schema[type + "Name"][value];
+            else
+                return value;
+        }
+        else if (Array.isArray(key)) {
+            let values = [];
+            for (let i = 0 ; i < key.length ; i++) {
+                values.push(this.getAttr(descriptor, key[i]));
+            }
+            return values.join(", ");
+        }
+        else {
+            return this.getAttr(descriptor, key);
+        }
+    }
+
+    setAttribute(layer) {
+        let layerType = layer.layerType();
+        let layerName = armnn.schema.LayerName[layerType];
+
+        let schema = this._metadata.getSchema(layerName);
+
+        // ignore unknown layer
+        if (!schema)
+            return;
+
+        let _layer = armnn.Node.castLayer(layer);
+
+        if (typeof schema["attributes"] != "undefined") {
+            for (let i = 0 ; i < schema.attributes.length ; i++) {
+                let attr = schema.attributes[i];
+
+                let value = this.packAttr(_layer, attr);
+                this._attributes.push(new armnn.Attribute(attr.name, attr.type, value));
+            }
+        }
+
+        if (typeof schema["inputs"] != "undefined") {
+            for (let i = 0 ; i < schema.inputs.length ; i++) {
+                let input = schema.inputs[i];
+                let value = _layer[input["src"]]();
+
+                if (value) {
+                    this._inputs.push(new armnn.Parameter(input["name"], '', null, value));
+                }
+            }
+        }
+
+        this._category = schema["category"];
+    }
+
+};
+
+armnn.Attribute = class {
+
+    constructor(name, type, value) {
+        this._name = name;
+        this._value = value;
+        this._visible = true;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get value() {
+        return this._value;
+    }
+
+    get visible() {
+        return this._visible == false ? false : true;
+    }
+};
+
+armnn.Parameter = class {
+
+    constructor(name, id, tensorInfo, initializer) {
+        this._name = name;
+        this._arguments = [ new armnn.Argument(id, tensorInfo, initializer) ];
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get visible() {
+        return true;
+    }
+
+    get arguments() {
+        return this._arguments;
+    }
+
+    static makeKey(layer_id, index) {
```

Comment:
`makeKey` is only used by `Graph` so it should move there and is private `_makeKey`.